docs/design/2020-06-27-non-recursive-common-table-expression.md

```diff
@@ -0,0 +1,85 @@
+
+# Proposal: Non-Recursive Common Table Expression
+
+- Author(s):     [Pingyu](https://github.com/pingyu) (Ping Yu)
+- Last updated:  2020-06-29
+- Discussion at: https://github.com/pingcap/tidb/issues/17472
+
+## Abstract
+
+This proposal proposes to support the __Non-recursive Common Table Expression (CTE)__.
+
+## Background
+
+From MySQL User Document, [13.2.15 WITH (Common Table Expressions)](https://dev.mysql.com/doc/refman/8.0/en/with.html):
```

Comment:
It's better to add the following information to this section:

- What is the common table expression
- How it's used in applications, what's the be benefit for application developers
- Issues which request to support common table expression

Reply:
Comment addressed, PTAL~
@zz-jason